tfhe/src/integer/server_key/radix_parallel/tests_unsigned/mod.rs

```diff
@@ -542,83 +542,31 @@ impl<F> CpuFunctionExecutor<F> {
     }
 }
 
-/// For unary function
-///
-/// Note, we don't do
-/// impl<F, I, O> TestExecutor<I, O> for CpuTestExecutor<F>
-/// where F: Fn(&ServerKey, I) -> O {}
-/// As it would conflict with other impls.
-///
-/// impl<F, I1, O> TestExecutor<(I,), O> for CpuTestExecutor<F>
-/// would be possible tho.
-impl<'a, F> FunctionExecutor<&'a RadixCiphertext, (RadixCiphertext, BooleanBlock)>
-    for CpuFunctionExecutor<F>
-where
-    F: Fn(&ServerKey, &RadixCiphertext) -> (RadixCiphertext, BooleanBlock),
-{
-    fn setup(&mut self, _cks: &RadixClientKey, sks: Arc<ServerKey>) {
-        self.sks = Some(sks);
-    }
+pub(crate) trait NotTuple {}
 
-    fn execute(&mut self, input: &'a RadixCiphertext) -> (RadixCiphertext, BooleanBlock) {
-        let sks = self.sks.as_ref().expect("setup was not properly called");
-        (self.func)(sks, input)
-    }
-}
+impl<T> NotTuple for &crate::integer::ciphertext::BaseRadixCiphertext<T> {}
 
-impl<'a, F> FunctionExecutor<&'a RadixCiphertext, RadixCiphertext> for CpuFunctionExecutor<F>
-where
-    F: Fn(&ServerKey, &RadixCiphertext) -> RadixCiphertext,
-{
-    fn setup(&mut self, _cks: &RadixClientKey, sks: Arc<ServerKey>) {
-        self.sks = Some(sks);
-    }
-
-    fn execute(&mut self, input: &'a RadixCiphertext) -> RadixCiphertext {
-        let sks = self.sks.as_ref().expect("setup was not properly called");
-        (self.func)(sks, input)
-    }
-}
-
-impl<'a, F> FunctionExecutor<&'a Vec<RadixCiphertext>, Option<RadixCiphertext>>
-    for CpuFunctionExecutor<F>
-where
-    F: Fn(&ServerKey, &Vec<RadixCiphertext>) -> Option<RadixCiphertext>,
-{
-    fn setup(&mut self, _cks: &RadixClientKey, sks: Arc<ServerKey>) {
-        self.sks = Some(sks);
-    }
+impl<T> NotTuple for &crate::integer::ciphertext::BaseSignedRadixCiphertext<T> {}
 
-    fn execute(&mut self, input: &'a Vec<RadixCiphertext>) -> Option<RadixCiphertext> {
-        let sks = self.sks.as_ref().expect("setup was not properly called");
-        (self.func)(sks, input)
-    }
-}
+impl<T> NotTuple for &mut crate::integer::ciphertext::BaseRadixCiphertext<T> {}
 
-/// Unary assign fn
-impl<'a, F> FunctionExecutor<&'a mut RadixCiphertext, ()> for CpuFunctionExecutor<F>
-where
-    F: Fn(&ServerKey, &'a mut RadixCiphertext),
-{
-    fn setup(&mut self, _cks: &RadixClientKey, sks: Arc<ServerKey>) {
-        self.sks = Some(sks);
-    }
+impl<T> NotTuple for &mut crate::integer::ciphertext::BaseSignedRadixCiphertext<T> {}
 
-    fn execute(&mut self, input: &'a mut RadixCiphertext) {
-        let sks = self.sks.as_ref().expect("setup was not properly called");
-        (self.func)(sks, input);
-    }
-}
+impl<T> NotTuple for &Vec<T> {}
 
-impl<'a, F> FunctionExecutor<&'a mut RadixCiphertext, RadixCiphertext> for CpuFunctionExecutor<F>
+/// For unary operations
+///
+/// Note, we need to `NotTuple` constraint to avoid conflicts with binary or ternary operations
```

Comment:
```rust
impl<F, I1, O> FunctionExecutor<I1, O> for CpuFunctionExecutor<F>
where
    F: Fn(&ServerKey, I1) -> O,
    I1: NotTuple,
{
    fn execute(&mut self, input: I1) -> O {
        let sks = self.sks.as_ref().expect("setup was not properly called");
        (self.func)(sks, input)
    }
}
```
conflicts with
```rust
impl<F, I1, I2 O> FunctionExecutor<I1, O> for CpuFunctionExecutor<F>
where
    F: Fn(&ServerKey, I1) -> O,
    I1: NotTuple,
{
    fn execute(&mut self, input: (I1, I2)) -> O {
        let sks = self.sks.as_ref().expect("setup was not properly called");
        (self.func)(sks, input)
    }
}
```

Because the first case `input: I1` conflicts with `input: (I1, I2)`